tests/subsys/fs/littlefs/src/test_lfs_basic.c

```diff
@@ -473,6 +475,198 @@ static int check_large(void)
+	zassert_equal(fs_open_ex(&file,
+				 testfs_path_init(&path, mp,
+						  CREATE,
+						  TESTFS_PATH_END),
+				FO_WRITE | FO_READ),
+		      -ENOENT,
+		      "verify non existing open for R/W failed");
+
+	/* Append on non existing should fail */
+	zassert_equal(fs_open_ex(&file,
+				 testfs_path_init(&path, mp,
+						  CREATE,
+						  TESTFS_PATH_END),
+				FO_APPEND | FO_WRITE | FO_READ),
+		      -ENOENT,
+		      "verify non existing append for R/W failed");
+
+	/* Open with FO_APPEND | FO_CREATE but with no FO_WRITE should fail */
+	zassert_equal(fs_open_ex(&file,
+				 testfs_path_init(&path, mp,
+						  CREATE,
+						  TESTFS_PATH_END),
+				FO_APPEND | FO_CREATE),
+		      -EINVAL,
+		      "verify non existing append or create, no write failed");
+
+	/* Open with FO_CREATE | FO_WRITE on non existing should create new */
+	zassert_equal(fs_open_ex(&file,
+				 testfs_path_init(&path, mp,
+						  CREATE,
+						  TESTFS_PATH_END),
+				FO_CREATE | FO_APPEND | FO_WRITE),
+		      0,
+		      "verify create for write failed");
+
+	/* Write one int to check if below FO_CREATE will not truncate it */
+	zassert_equal(fs_write(&file, &c, sizeof(c)), sizeof(c),
+		      "verify write should be possible");
+
+	zassert_equal(fs_seek(&file, 0, FS_SEEK_END), 0,
+		      "failed to move to the end of file");
+
+	ret = fs_tell(&file);
+	zassert_equal(ret, sizeof(c),
+		      "expected file size %d, got %d", sizeof(c), ret);
+
+	zassert_equal(fs_close(&file),
+		      0,
+		      "verify close after create failed");
+
+	/* FO_CREATE on existing file should not destroy it */
+	zassert_equal(fs_open_ex(&file,
+				 testfs_path_init(&path, mp,
+						  CREATE,
+						  TESTFS_PATH_END),
+				 FO_CREATE | FO_WRITE),
+		      0,
+		      "verify open existing for write failed");
+
+	/* There should be one int */
+	zassert_equal(fs_seek(&file, 0, FS_SEEK_END), 0,
+		      "failed to move to the end of file");
+
+	ret = fs_tell(&file);
+	zassert_equal(ret, sizeof(c),
+		      "expected file size %d, got %d", sizeof(c), ret);
+
+	zassert_equal(fs_close(&file),
+		      0,
+		      "verify close after create failed");
+
+	/* Check if above FO_CREATE left contents of the file */
+	/* Open existing with no flags should be possible */
+	zassert_equal(fs_open_ex(&file,
+				 testfs_path_init(&path, mp,
+						  CREATE,
+						  TESTFS_PATH_END),
+				 0),
+		      0,
+		      "verify open existing failed");
+
+	/* NOTE: The Zephyr fs_open_ex flags allow opening existing files
+	 * with READ/WRITE access, as it happens in fs_open_ex above, but
+	 * the LittleFS assumes READ/WRITE access by default if no access
+	 * is selected which same as requesting READ ONLY and WRITE ONLY
+	 * access at the same time (bug?).
+	 * From Zephyr API point of view below tests should fail but they
+	 * will succeed, which is expected (?) behaviour of LittleFS;
+	 * that is why this test is left here commented out.
```

Comment:
If I'm understanding you right, you are saying that littlefs requires flags to be either `LFS_O_RDONLY`, `LFS_O_WRONLY`, or `LFS_O_RDWR`, and passing a flag value of 0 is treated the same as `LFS_O_RDWR`.  If that's true, shouldn't it be the job of `lfs_flags_from_zephyr` to never pass a flag value of 0 to lfs?

Reply:
`lfs_flags_from_zephyr ` is supposed to translate flags to obtain, from fs backend, specific behavior. I guess that the READONLY would be closest to expected.